conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/generate/ConjureUndertowEndpointsGenerator.java

```diff
@@ -755,6 +757,26 @@ private static Optional<CodeBlock> getSafeLogging(
         }
     }
 
+    private static final ClassName LIST_NAME = ClassName.get(List.class);
+    private static final ClassName IMMUTABLE_LIST_NAME = ClassName.get(ImmutableList.class);
+    private static final ClassName SET_NAME = ClassName.get(Set.class);
+    private static final ClassName IMMUTABLE_SET_NAME = ClassName.get(ImmutableSet.class);
+
+    private static TypeName immutableCollection(TypeName input) {
+        if (input instanceof ParameterizedTypeName) {
+            ParameterizedTypeName parameterized = (ParameterizedTypeName) input;
+            if (LIST_NAME.equals(parameterized.rawType)) {
```

Comment:
Perhaps `Collection` and maybe `Iterable` should become `ImmutableList`?

Reply:
Maybe Map as well, but I'd hope that's less common and may not be necessary.

Reply:
I've added `Collection`, `Map`, and a couple more collection types.

`Iterable` feels a bit weird, I'd slightly prefer to not support that.